lib/src/test/java/io/ably/lib/test/rest/RestAuthAttributeTest.java

```diff
@@ -0,0 +1,109 @@
+     * <p>
+     * Spec: RSA10g
+     * </p>
+     */
+    @Test
+    public void auth_stores_options_params() {
+        try {
+            /* init custom TokenParams */
+            Capability capability1 = new Capability();
+            capability1.addResource("testchannel", "subscribe");
+            TokenParams tokenParams = new TokenParams() {{
+                ttl = 1000L;
+                clientId = "firstClientId";
+                capability = capability1.toString();
+            }};
+
+            /* init custom AuthOptions */
+            Setup.TestVars testVars = Setup.getTestVars();
+            AuthOptions authOptions = new AuthOptions() {{
+                authCallback = new TokenCallback() {
+                    private AblyRest ably = new AblyRest(testVars.createOptions(testVars.keys[0].keyStr));
+
+                    @Override
+                    public Object getTokenRequest(TokenParams params) throws AblyException {
+                        return ably.auth.requestToken(params, null);
+                    }
+                };
+                authUrl = "auth_url_1";
+                token = "test_token_1";
+                key = testVars.keys[1].keyStr;
+                tokenDetails = new TokenDetails();
+                authHeaders = new Param[]{new Param("X-Header-Param", "1")};
+                authParams = new Param[]{new Param("X-Auth-Param", "1")};
+            }};
+
+            /* authorise with custom options */
+            ably.auth.authorise(authOptions, tokenParams);
+
+            /* get stored TokenParams */
+            TokenParams storedTokenParams = ably.auth.getTokenParams();
+
+            /* Verify that,
+             * storedTokenParams isn't null,
+             * and storedTokenParams equals custom tokenParams */
+            assertNotNull(storedTokenParams);
+            assertEquals(storedTokenParams.ttl, tokenParams.ttl);
+            assertEquals(storedTokenParams.clientId, tokenParams.clientId);
+            assertEquals(storedTokenParams.capability, tokenParams.capability);
+
+            /* get stored AuthOptions */
+            AuthOptions storedAuthOptions = ably.auth.getAuthOptions();
+
+            /* Verify that,
+             * storedAuthOptions isn't null,
+             * and storedAuthOptions equals custom authOptions */
+            assertNotNull(storedAuthOptions);
+            assertEquals(storedAuthOptions.authCallback, authOptions.authCallback);
+            assertEquals(storedAuthOptions.authUrl, authOptions.authUrl);
+            assertEquals(storedAuthOptions.token, authOptions.token);
+            assertEquals(storedAuthOptions.key, authOptions.key);
+            assertEquals(storedAuthOptions.tokenDetails, authOptions.tokenDetails);
+            assertArrayEquals(storedAuthOptions.authHeaders, authOptions.authHeaders);
+            assertArrayEquals(storedAuthOptions.authParams, authOptions.authParams);
```

Comment:
Note that call authorise will not actually do anything if a token exists, so you need the first token to expire so that a new one has to be issued and you need to assert that the new one is not the same as the old in your test.